test/lib/rubycritic/generators/json_report_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+require 'rubycritic/core/analysed_modules_collection'
+require 'rubycritic/generators/json_report'
+require 'json'
+
+describe RubyCritic::Generator::JsonReport do
+  describe '#generate_report' do
+    before do
+      FileUtils.rm('test/samples/report.json') if File.file?('test/samples/report.json')
+      create_analysed_modules_collection
+      generate_report
```

Comment:
I find it confusing that the examples below just contain "assert". I think we should move the report generation into the examples. Using a fake file system like suggested above will also make this easier since you don't have to clean up afterwards. Meaning that you can just move "generate_report" to each examples with duplicating the set up everywhere.